rtpsniff.c

```diff
@@ -47,26 +47,28 @@ int main(int argc, char const *const *argv) {
 
     /* Try initialization */
     errbuf[0] = '\0';
-    strncat(errbuf, "Not enough arguments", PCAP_ERRBUF_SIZE - 1);
-    if (argc != 4 ||
-	    ((handle = pcap_create(argv[1], errbuf)) == NULL) ||
-	    (pcap_set_snaplen(handle, sniff_snaplen()) != 0) ||
-	    (pcap_set_timeout(handle, 1000) != 0) ||
+    if (argc != 4) {
+	strncat(errbuf, "Not enough arguments", PCAP_ERRBUF_SIZE - 1);
+	exit_error(errbuf);
+    }
+    if ((handle = pcap_create(argv[1], errbuf)) == NULL)
+	exit_error(errbuf);
+    if ((pcap_set_snaplen(handle, sniff_snaplen()) != 0) ||
+	(pcap_set_timeout(handle, 1000) != 0) ||
 	    /* frame size is 128 for our snaplen,
 	     * we need to hold MAX_KPPS*1000*128 bytes if we poll and flush
 	     * every second. add a little extra for kicks.
 	     * If you set this too low, the "packets dropped by kernel" total
 	     * will increase dramatically. Note that that figure will always
 	     * show some packets because of startup/shutdown misses. */
-	    (pcap_set_buffer_size(handle, (bufsize = atoi(argv[2]) * 1024 * 192)) != 0) ||
-	    (pcap_activate(handle) != 0) ||
-	    (pcap_compile(handle, &fp, argv[3], 0, PCAP_NETMASK_UNKNOWN) == -1) ||
-	    (pcap_setfilter(handle, &fp) == -1)) {
-	fprintf(stderr, "rtpsniff: Initialization failed or bad command line "
-		        "options. See -h for help:\n%s\n", errbuf);
+	(pcap_set_buffer_size(handle, (bufsize = atoi(argv[2]) * 1024 * 192)) != 0) ||
+	(pcap_activate(handle) != 0) ||
+	(pcap_activate(handle) != 0) ||
+	(pcap_compile(handle, &fp, argv[3], 0, PCAP_NETMASK_UNKNOWN) == -1) ||
+	(pcap_setfilter(handle, &fp) == -1)) {
 	if (handle)
 	    pcap_close(handle);
-	return 0;
+	exit_error(pcap_geterr(handle));
```

Comment:
Ik vraag me af of het legaal is om handle te lezen nadat je 'm geclosed hebt. Zou zomaar kunnen van niet.